keras_cv/layers/preprocessing/grid_mask.py

```diff
@@ -123,8 +123,12 @@ def _compute_mask(self, image_height, image_width):
 
         if self.fill_mode == "constant":
             mask = tf.fill([mask_h, mask_w], self.fill_value)
-        else:
+        elif self.fill_mode == "gaussian_noise":
             mask = tf.cast(tf.random.normal([mask_h, mask_w]), dtype=tf.int32)
+        else:
+            raise ValueError(
+                "Unsupported fill_mode.  `fill_mode` should be 'constant' or 'gaussian_noise'."
+            )
```

Comment:
@LukeWood the else check here may not be needed as we already check the input params in the `init` method, within 

```python 
super().__init__(**kwargs)
self._check_input_range(ratio, fill_mode, fill_value)

if fill_mode.lower() not in ["constant", "gaussian_noise"]:
     raise ValueError(
          f"fill_mode should be either 'constant', or 'gaussian_noise' ]. Got {fill_mode}"
)
````